engine/src/cmd/upgradeable_unit.cpp

```diff
@@ -141,6 +142,14 @@ bool UpgradeableUnit::UpgradeMounts(const Unit *up,
         return true;
     }
 
+    // there needs to be some mounts to be able to mount to
+    if (num_mounts == 0) {
+        // would be nice to make this more meaningful but that's a little harder given
+        // the casting of `unit` from `this`.
+        VS_LOG(debug, "No mounts to attach to.");
+        return false;
+    }
```

Comment:
I kept having crashes due to L157 doing a Divide by Zero, so put something in to protect against that.